packages/react-dom/src/__tests__/ReactFresh-test.internal.js

```diff
@@ -0,0 +1,1421 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+/* eslint-disable no-for-of-loops/no-for-of-loops */
+
+'use strict';
+
+let React;
+let ReactDOM;
+let act;
+
+describe('ReactFresh', () => {
+  let container;
+  let familiesByID;
+  let familiesByType;
+  let newFamilies;
+  let updatedFamilies;
+  let performHotReload;
+  let signaturesByType;
+
+  beforeEach(() => {
+    let enableHotReloading;
+    let lastRoot;
+    global.__REACT_DEVTOOLS_GLOBAL_HOOK__ = {
+      supportsFiber: true,
```

Comment:
If this sets `supportsHotReloading` to `true`, then we can avoid the `enableHotReloading` stuff.

Reply:
I wanted three things: 

1. No overhead if hot reloading is supported
2. A way for the hot runtime to inject family map into React
3. A way for the hot runtime to plug into global Hook

Note hot runtime isn't part of React DevTools. It just reuses the same mechanism. So it'll have to play nicely with DevTools itself. How does that happen?

If we make this `supportsHotReloading: true`, I supposed it'll be `false` by default in DevTools, and then hot runtime would mutate it to true? Note it would still need to override `inject` in order to gain access to `scheduleHotUpdate`, as well as `onCommitFiberRoot` to track roots.

How does the family map get injected into React? A separate method like `injectHotReloadingMap`?

Maybe family map could be a part of first `scheduleHotReload` call. While the map would have to be the same every time, this would mean we may zero price until you actually use the feature the first time.

Reply:
Addressed.